src/app/leverage/components/stats/leverage-selector-container.tsx

```diff
@@ -1,160 +1,168 @@
-//   isPositive: boolean
-// }
+type LeverageRatioResponse = {
+  ratio: number
+  strategy: string
+}
 
-// const ratios: LeverageRatio[] = [
-//   {
-//     icon: getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//     ratio: '2x',
-//     // TODO: merge other earn PR first or cherry-pick here?
-//     networks: [
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       // '/assets/ethereum-network-logo.svg',
-//       // '/assets/arbitrum-network-logo.svg',
-//       // '/assets/base-network-icon.svg',
-//     ],
-//     leverage: '1.94x',
-//     isPositive: true,
-//   },
-//   {
-//     icon: getTokenByChainAndSymbol(arbitrum.id, 'ETH3X').logoURI,
-//     ratio: '3x',
-//     networks: [
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       // '/assets/ethereum-network-logo.svg',
-//       // '/assets/arbitrum-network-logo.svg',
-//     ],
-//     leverage: '2.96x',
-//     isPositive: true,
-//   },
-//   {
-//     icon: getTokenByChainAndSymbol(arbitrum.id, 'iETH1X').logoURI,
-//     ratio: '-1x',
-//     networks: [
-//       getTokenByChainAndSymbol(arbitrum.id, 'ETH2X').logoURI,
-//       // '/assets/ethereum-network-logo.svg'
-//     ],
-//     leverage: '-1.10x',
-//     isPositive: false,
-//   },
-// ]
+export function LeverageSelectorContainer() {
+  const { chainId } = useNetwork()
+  const { isConnected } = useWallet()
+  const { indexToken, leverageType, market } = useLeverageToken()
+  const {
+    data: { token },
+    isFetchingQuickStats,
+  } = useQuickStats(market, indexToken)
 
-// function LeverageRatioItem({ item }: { item: LeverageRatio }) {
-//   return (
-//     <div className='border-ic-gray-600 flex items-center justify-between border-t px-4 py-3 first:border-t-0'>
-//       <div className='flex items-center gap-2'>
-//         <Image
-//           src={item.icon}
-//           alt={`${item.ratio} leverage`}
-//           height={16}
-//           width={16}
-//         />
-//         <span className='text-ic-white text-xs font-medium'>{item.ratio}</span>
-//       </div>
-//       <div className='flex gap-1'>
-//         {item.networks.map((network, idx) => (
-//           <Image
-//             key={idx}
-//             src={network}
-//             alt={'Network Icon'}
-//             height={12}
-//             width={12}
-//           />
-//         ))}
-//       </div>
-//       <span className={'text-ic-white text-right text-xs font-medium'}>
-//         {item.leverage}
-//       </span>
-//     </div>
-//   )
-// }
+  const { data } = useQuery({
+    enabled: typeof chainId === 'number',
+    gcTime: 5 * 60 * 1000,
+    refetchOnWindowFocus: false,
+    queryKey: ['leverage-ratio', market, chainId],
+    queryFn: async () => {
+      const res = await fetch(
```

Comment:
possible spot for kubb I think, In case its a non-authenticated route?

Reply:
p.s. this is using kubb inside the next.js route